packages/ckeditor5-image/src/image/replaceimagesourcecommand.ts

```diff
@@ -43,10 +61,30 @@ export default class ReplaceImageSourceCommand extends Command {
 	 */
 	public override execute( options: { source: string } ): void {
 		const image = this.editor.model.document.selection.getSelectedElement()!;
+
 		this.editor.model.change( writer => {
 			writer.setAttribute( 'src', options.source, image );
-			writer.removeAttribute( 'srcset', image );
-			writer.removeAttribute( 'sizes', image );
+
+			this._cleanupImage( writer, image );
 		} );
 	}
+
+	/**
+	 * Cleanup some image attributes.
+	 */
+	private _cleanupImage( writer: Writer, image: Element ) {
```

Comment:
If this was `public` and `decorate`d, you wouldn't need to maintain `_imageCallbacks`.